hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java

```diff
@@ -491,12 +492,13 @@ private void handleAppSubmitEvent(AbstractDelegationTokenRenewerAppEvent evt)
     Set<DelegationTokenToRenew> tokenList = new HashSet<DelegationTokenToRenew>();
     boolean hasHdfsToken = false;
     for (Token<?> token : tokens) {
+      AtomicLong tokenExpiredTime = new AtomicLong(now);
       if (token.isManaged()) {
         if (token.getKind().equals(HDFS_DELEGATION_KIND)) {
           LOG.info(applicationId + " found existing hdfs token " + token);
           hasHdfsToken = true;
         }
-        if (skipTokenRenewal(token)) {
+        if (skipTokenRenewal(token, tokenExpiredTime)) {
```

Comment:
The var of `tokenExpiredTime` will be changed in the method of `skipTokenRenewal`, however we can't found it according to its method name directly. Maybe `skipTokenRenewal` should be renamed to `skipTokenRenewalAndUpdateExpiredTime`.